src/image_builder.py

```diff
@@ -154,109 +155,177 @@ def image_builder(buildspec):
+                dockerfile=os.path.join(os.sep, "Dockerfile.multipart"),
+                repository=image_repo_uri,
+                tag=image_tag,
+                to_build=image_config["build"],
+                stage=constants.SECOND_STAGE,
+                context=None,
+            )
 
-    # In the context of the ThreadPoolExecutor each instance of image.build submitted
-    # to it is executed concurrently in a separate thread.
-    with concurrent.futures.ThreadPoolExecutor(max_workers=10) as executor:
-        # Standard images must be built before example images
-        # Example images will use standard images as base
-        standard_images = [image for image in IMAGES if "example" not in image.name.lower()]
-        example_images = [image for image in IMAGES if "example" in image.name.lower()]
+        FORMATTER.separator()
 
-        for image in standard_images:
-            THREADS[image.name] = executor.submit(image.build)
+        FIRST_STAGES_IMAGES.append(first_stage_image_object)
+        SECOND_STAGES_IMAGES.append(second_stage_image_object)
 
-        # the FORMATTER.progress(THREADS) function call also waits until all threads have completed
-        FORMATTER.progress(THREADS)
+    FORMATTER.banner("DLC")
+    #FORMATTER.title("Status")
+    
+    # Standard images must be built before example images
+    # Example images will use standard images as base
+    first_stage_standard_images = [image for image in FIRST_STAGES_IMAGES if "example" not in image.name.lower()]
+    second_stage_standard_images = [image for image in SECOND_STAGES_IMAGES]
+    
+    example_images = [image for image in FIRST_STAGES_IMAGES if "example" in image.name.lower()]
+    #needs to be reconfigured
+    #ALL_IMAGES = first_stage_standard_images + second_stage_standard_images + example_images
+
+    #first stage build
+    FORMATTER.banner("First Stage Build")
+    build_images(first_stage_standard_images)
 
-        for image in example_images:
-            THREADS[image.name] = executor.submit(image.build)
+    """
+    Run safety on first stage image and store the ouput file locally
+    """
 
-        # the FORMATTER.progress(THREADS) function call also waits until all threads have completed
-        FORMATTER.progress(THREADS)
+    FORMATTER.banner("Second Stage Build")
+    build_images(SECOND_STAGES_IMAGES)
+    
+    #second stage build
+    if second_stage_standard_images:
+        FORMATTER.banner("Second Stage Build")
+        build_images(second_stage_standard_images)
+    
+    push_images(second_stage_standard_images)
+
+    #example image build
+    build_images(example_images)
+    push_images(example_images)
+
+    #After the build, display logs/sumary for all the images.
+
+    show_build_logs(ALL_IMAGES)
+    show_build_summary(ALL_IMAGES)
+    is_any_build_failed, is_any_build_failed_size_limit = show_build_errors(ALL_IMAGES)
+
+    #change logic here. upload metrics only for the second stage image
```

Comment:
If I look at `push_image_metrocs` function in ~/deep-learning-containers/src/metrics.py  I observe that build time is generally `image.summary["end_time"] - image.summary["start_time"]`. If we only push the metrics for the second stage image, we will not include the amount of time it takes to build the entire first stage image. We should account for that somehow. 

Also, I feel that we should not build second stage images in case the first stage images fail. We should add that logic handling in the code.